src/index.ts

```diff
@@ -167,6 +172,81 @@ export class UltravoxSession {
     this.sendData({ type: 'input_text_message', text });
   }
 
+  mute(roles: RoleString | RoleString[] | Set<RoleString> | Role | Role[] | Set<Role>): void {
+    const roleSet = this.getRoleSet(roles);
+
+    for (const role of roleSet) {
+      switch (role) {
+        case Role.AGENT:
+          if (!this.room?.remoteParticipants) {
+            throw new Error('Cannot set isAgentMuted.');
+          }
+          this.isAgentMuted = true;
+          this.room.remoteParticipants.forEach((rp) => {
+            rp.audioTrackPublications.forEach((atp) => {
+              atp.track?.setMuted(this.isAgentMuted);
+            });
+          });
+          break;
+
+        case Role.USER:
+          if (!this.room?.localParticipant) {
+            throw new Error('Cannot set isCallerMuted.');
+          }
+          this.isUserMuted = true;
+          this.room.localParticipant.setMicrophoneEnabled(!this.isUserMuted);
+          break;
+
+        default:
+          break;
+      }
+    }
+  }
+
+  unmute(roles: RoleString | RoleString[] | Set<RoleString> | Role | Role[] | Set<Role>): void {
+    const roleSet = this.getRoleSet(roles);
+
+    for (const role of roleSet) {
+      switch (role) {
+        case Role.AGENT:
+          if (!this.room?.remoteParticipants) {
+            throw new Error('Cannot set isAgentMuted.');
+          }
+          this.isAgentMuted = false;
+          this.room.remoteParticipants.forEach((rp) => {
+            rp.audioTrackPublications.forEach((atp) => {
+              atp.track?.setMuted(this.isAgentMuted);
+            });
+          });
+          break;
+
+        case Role.USER:
+          if (!this.room?.localParticipant) {
+            throw new Error('Cannot set isCallerMuted.');
+          }
+          this.isUserMuted = false;
+          this.room.localParticipant.setMicrophoneEnabled(!this.isUserMuted);
```

Comment:
If you're interested in matching the Flutter and Python updates I just shipped, we should add user_muted and agent_muted events also (and also remove UltravoxSessionState by making UltravoxSession the EventTarget and rename events).  If you want to punt on that, I can pick it up later too.

Reply:
ok i will do this in a separate PR